tests/testing_four_c_simulation.py

```diff
@@ -598,26 +604,113 @@ def test_four_c_simulation_dirichlet_boundary_to_neumann_boundary(self):
         then apply those as Neumann boundaries.
         """
 
-        def create_model(n_steps):
-            """Create the cantilver model."""
-
-            mpy.set_default_values()
-            input_file = InputFile()
-            set_header_static(input_file, time_step=0.5, n_steps=n_steps)
-            input_file.add(
-                "--IO\nOUTPUT_BIN yes\nSTRUCT_DISP yes", option_overwrite=True
+        # Create and run the initial simulation.
+        initial_simulation, beam_set = create_cantilver_model(n_steps=2)
+        initial_simulation.add(
+            BoundaryCondition(
+                beam_set["start"],
+                "NUMDOF 9 ONOFF 1 1 1 1 1 1 0 0 0 VAL 0 0 0 0 0 0 0 0 0 FUNCT 0 0 0 0 0 0 0 0 0",
+                bc_type=mpy.bc.dirichlet,
             )
-            ft = Function("COMPONENT 0 SYMBOLIC_FUNCTION_OF_SPACE_TIME t")
-            input_file.add(ft)
-            mat = MaterialReissner(youngs_modulus=100.0, radius=0.1)
-            beam_set = create_beam_mesh_line(
-                input_file, Beam3rHerm2Line3, mat, [0, 0, 0], [2, 0, 0], n_el=10
+        )
+        initial_simulation.add(
+            BoundaryCondition(
+                beam_set["end"],
+                "NUMDOF 9 ONOFF 1 1 1 0 0 0 0 0 0 VAL -0.2 1.5 1 0 0 0 0 0 0 FUNCT 1 1 1 0 0 0 0 0 0 TAG monitor_reaction",
+                bc_type=mpy.bc.dirichlet,
             )
+        )
+        initial_simulation.add(
+            """
+            --IO/MONITOR STRUCTURE DBC
+            PRECISION_FILE         10
+            PRECISION_SCREEN       5
+            FILE_TYPE              csv
+            WRITE_HEADER           yes
+            INTERVAL_STEPS         1
+            """
+        )
+
+        # Check the input file
+        compare_test_result(
+            self,
+            initial_simulation.get_string(check_nox=False, header=False),
+            additional_identifier="initial",
+        )
+
+        # Run the simulation in 4C
+        self.run_four_c_test(
+            "dbc_to_nbc_initial", initial_simulation, delete_files=False
+        )
+
+        # Create and run the second simulation.
+        restart_simulation, beam_set = create_cantilver_model(n_steps=21)
+        restart_simulation.add(
+            BoundaryCondition(
+                beam_set["start"],
+                "NUMDOF 9 ONOFF 1 1 1 1 1 1 0 0 0 VAL 0 0 0 0 0 0 0 0 0 FUNCT 0 0 0 0 0 0 0 0 0",
+                bc_type=mpy.bc.dirichlet,
+            )
+        )
+        function_nbc = Function(
+            """SYMBOLIC_FUNCTION_OF_TIME nbc_value
+            VARIABLE 0 NAME nbc_value TYPE linearinterpolation NUMPOINTS 2 """
+            "TIMES 1.0 11.0 VALUES 1.0 0.0"
+        )
+        restart_simulation.add(function_nbc)
+        all_dbc_monitor_values_to_input(
+            restart_simulation,
+            os.path.join(
+                testing_temp,
```

Comment:
Where does all this come from? Is it possible that you added the same test again (the two tests look like they have the same name)